### PR TITLE
Reduce ResettableDictionary lookups

### PR DESCRIPTION
**Pull request by benaadams**
## Changes

- Use return value from first lookup rather than looking up each time

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [x] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/Nethermind/Nethermind.State/StateProvider.cs [hunk 1]
```diff
@@ -761,12 +764,15 @@ private void IncrementChangePosition()
...
 
-        private void SetupCache(Address address)
+        private Stack<int> SetupCache(Address address)
         {
-            if (!_intraBlockCache.ContainsKey(address))
+            ref Stack<int>? value = ref _intraBlockCache.GetValueRefOrAddDefault(address, out bool exists);
+            if (!exists)
             {
-                _intraBlockCache[address] = new Stack<int>();
+                value = new Stack<int>();
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.State/StateProvider.cs:772]: this new dictionary is not added to _intraBlockCache

**Review by LukaszRozmej — CHANGES_REQUESTED** [review]

File: src/Nethermind/Nethermind.State/PartialStorageProviderBase.cs [hunk 2]
```diff
@@ -261,12 +262,15 @@ protected void IncrementChangePosition()
...
         /// <param name="cell"></param>
-        protected void SetupRegistry(in StorageCell cell)
+        protected StackList<int> SetupRegistry(in StorageCell cell)
         {
-            if (!_intraBlockCache.ContainsKey(cell))
+            ref StackList<int>? value = ref _intraBlockCache.GetValueRefOrAddDefault(cell, out bool exists);
+            if (!exists)
             {
-                _intraBlockCache[cell] = new StackList<int>();
+                value = new StackList<int>();
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.State/PartialStorageProviderBase.cs:270]: same here

File: src/Nethermind/Nethermind.State/StateProvider.cs (hunk 1 above)
Comment (benaadams) [reply src/Nethermind/Nethermind.State/StateProvider.cs:772]: It is as `value` is a `ref`

File: src/Nethermind/Nethermind.State/PartialStorageProviderBase.cs (hunk 2 above)
Comment (benaadams) [reply src/Nethermind/Nethermind.State/PartialStorageProviderBase.cs:270]: It is as `value` is a `ref`

**Review by LukaszRozmej — APPROVED** [review]

**Pull request merged**